app/services/course_service.py

```diff
@@ -60,4 +65,18 @@ async def get_instructor_gitea_organization_name(self) -> str:
     
     async def get_master_repository_name(self) -> str:
         course = await self.get_course()
-        return f"{ course.name }-class-master-repo"
\ No newline at end of file
+        return f"{ course.name }-class-master-repo"
+    
+    async def update_course(self, name: str, total_students: int, start_at: datetime, end_at: datetime) -> CourseModel:
+        print('bar')
+        try:
+            course = await self.get_course()
+            course.name = name
+            course.total_students = total_students
+            course.start_at = start_at
+            course.end_at = end_at
+            self.session.commit()
+            return course
+        except NoCourseExistsException:
```

Comment:
also a feature I recently learned of when raising a different exception from an existing one is that you can do `raise from`
```
except Exception as exception:
    raise NewException from exception
```
though I'm not sure exactly how it works or if it would work for our exceptions. requires some testing

Reply:
yeah agreed, initially I had added it to catch the exception in the case of creating a course for the first time. But I am now handling that logic in `lms_sync_script`. Removing this exception from the course service